api/src/pcapi/core/offers/repository.py

```diff
@@ -724,3 +725,27 @@ def search_offers_by_filters(
             offers = offers.filter(sa.func.unaccent(Offer.name).ilike(f"%{name_query}%")).limit(limit)
 
     return offers.distinct()
+
+
+def exclude_offers_from_inactive_venue_provider(query: BaseQuery) -> BaseQuery:
+    return (
+        query.join(
+            providers_models.Provider,
+            Offer.lastProviderId == providers_models.Provider.id,  # pylint: disable=comparison-with-callable
+            isouter=True,
+        )
```

Comment:
(détails / question de goût) Je pense même qu'on peut simplifier comme suit : 

```suggestion
        query.outerjoin(Offer.lastProvider)
```

Reply:
Ça n'a pas l'air de fontionner, je pense que c'est parce que Venue peut être liée à Provider aussi

```
sqlalchemy.exc.AmbiguousForeignKeysError: Can't determine join between 'Join object on Join object on Join object on offer(4494854656) and venue(4493604512)(5767600672) and offerer(4494083744)(5767600480) and user_offerer(4494090944)' and 'provider'; tables have more than one foreign key constraint relationship between them. Please specify the 'onclause' of this join explicitly.
```

Reply:
Ah ? Bizarre. J'étais curieux, donc j'ai testé ta branche avec le diff ci-dessous. Je passe bien dans la fonction `exclude_offers_from_inactive_venue_provider()` et le test `ExcludeOffersFromInactiveVenueProviderTest` passe.

```diff
 def exclude_offers_from_inactive_venue_provider(query: BaseQuery) -> BaseQuery:
     return (
-        query.join(
-            providers_models.Provider,
-            Offer.lastProviderId == providers_models.Provider.id,  # pylint: disable=comparison-with-callable
-            isouter=True,
-        )
+        query.outerjoin(Offer.lastProvider)
         .join(
             providers_models.VenueProvider,
             sa.and_(
```

Reply:
Ah bien vu j'avais écrit `outerjoin(providers_models.Provider)` plutôt que `outerjoin(Offer.lastProvider)`